server/webhook_jira.go

```diff
@@ -52,9 +52,20 @@ func (jwh *JiraWebhook) expandIssue(p *Plugin, instanceID types.ID) error {
 			}
 
 			jwh.Issue = *issue
-		} else if _, ok := instance.(*cloudOAuthInstance); ok {
+		} else if instance, ok := instance.(*cloudOAuthInstance); ok {
 			mmUserID, err := p.userStore.LoadMattermostUserID(instanceID, jwh.Comment.Author.AccountID)
 			if err != nil {
+				if instance.JWTInstance != nil {
```

Comment:
I think this would be easier to read if we returned early here with `instance.JWTInstance == nil ` instead of `instance.JWTInstance != nil `. Also added comments and an error message

```go
if err != nil {
    // User is not connected, so we try to fall back to JWT bot
    if instance.JWTInstance == nil {
        return errors.Wrap(err, "Cannot create subscription posts for this comment as the Jira comment author is not connected to Mattermost.")
    }

    // Fetch issue details with bot JWT bot
    var issue *jira.Issue
    issue, err = p.getIssueDataForCloudWebhook(instance.JWTInstance, jwh.Issue.ID)
    if err != nil {
        return errors.Wrap(err, "failed to getIssueDataForCloudWebhook using bot account")
    }

    jwh.Issue = *issue
    return nil
}
```